pymc/variational/opvi.py

```diff
@@ -370,10 +370,11 @@ def step_function(
             more_replacements=more_replacements,
             total_grad_norm_constraint=total_grad_norm_constraint,
         )
+        seed = self.approx.rng.integers(2**30, dtype=np.int64)
         if score:
-            step_fn = compile_pymc([], updates.loss, updates=updates, **fn_kwargs)
+            step_fn = compile_pymc([], updates.loss, updates=updates, random_seed=seed, **fn_kwargs)
```

Comment:
Specifically `random_seed` in `compile_pymc` won't do anything with `MRG` `RNG` state variables. We don't actually want to do that until we sort out #6277, otherwise two distinct minibatch RNGs would always be given different initial seed values.

Reply:
Once minibatching is fixed, it should become valid, right?

Reply:
Does it make sense to fix in one PR and get rid of MRG?

Reply:
Yes it does!